build/devenv/environment.go

```diff
@@ -283,153 +285,195 @@ func NewEnvironment() (in *Cfg, err error) {
 			return nil, err
 		}
 	}
+	///////////////////////////
+	// END: Deploy contracts //
+	///////////////////////////
+
+	////////////////////////////
+	// Start: Launch CL Nodes //
+	////////////////////////////
 
+	timeTrack.Record("[infra] deploying CL nodes")
+	err = launchCLNodes(ctx, in, impls)
+	if err != nil {
+		return nil, fmt.Errorf("failed to launch CL nodes: %w", err)
+	}
 	timeTrack.Record("[infra] deployed CL nodes")
-	timeTrack.Record("[changeset] deployed product contracts")
-
-	if hasCLNodeService { //nolint:nestif // large block needed for clarity, refactor as a cl node component later
-		clChainConfigs := make([]string, 0)
-		clChainConfigs = append(clChainConfigs, CommonCLNodesConfig)
-		for i, impl := range impls {
-			var clChainConfig string
-			clChainConfig, err = impl.ConfigureNodes(ctx, in.Blockchains[i])
-			if err != nil {
-				return nil, fmt.Errorf("failed to deploy local networks: %w", err)
-			}
-			clChainConfigs = append(clChainConfigs, clChainConfig)
-		}
-		allConfigs := strings.Join(clChainConfigs, "\n")
-		for _, nodeSpec := range in.NodeSets[0].NodeSpecs {
-			nodeSpec.Node.TestConfigOverrides = allConfigs
-		}
-		Plog.Info().Msg("Nodes network configuration is generated")
 
-		timeTrack.Record("[changeset] configured nodes network")
-		_, err = ns.NewSharedDBNodeSet(in.NodeSets[0], nil)
+	//////////////////////////
+	// End: Launch CL Nodes //
+	//////////////////////////
+
+	///////////////////////////////////////
+	// Start: Launch standalone services //
+	///////////////////////////////////////
+
+	_, err = launchStandaloneExecutor(in)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create standalone executor: %w", err)
+	}
+
+	_, err = launchStandaloneVerifiers(in)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create standalone verifiers: %w", err)
+	}
+
+	/////////////////////////////////////
+	// End: Launch standalone services //
+	/////////////////////////////////////
+
+	timeTrack.Print()
+	if err = PrintCLDFAddresses(in); err != nil {
+		return nil, err
+	}
+
+	return in, Store(in)
+}
+
+// launchCLNodes encapsulates the logic required to launch the core node. It may be better to wrap this in a service.
+func launchCLNodes(ctx context.Context, in *Cfg, impls []cciptestinterfaces.CCIP17ProductConfiguration) error {
+	// Exit early, there are no nodes configured.
+	if len(in.NodeSets) == 0 {
+		return nil
+	}
+
+	hasAService := false
+	for _, ver := range in.Verifier {
+		hasAService = hasAService || (ver.Mode == services.CL)
+	}
+
+	if in.Executor != nil {
+		hasAService = hasAService || (in.Executor.Mode == services.CL)
+	}
+
+	// Exit early, there are no services configured to deploy on a CL node.
+	if !hasAService {
+		return nil
+	}
```

Comment:
The `launchCLNodes` function simply wrapped the existing code in a function. The only new logic is this block at the top which skips starting a node if nothing is configured to run on the node or no nodes are configured.